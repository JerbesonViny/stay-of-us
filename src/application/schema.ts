import { buildSchemaSync } from "type-graphql";

export const schema = buildSchemaSync({
  resolvers: [__dirname + "/**/*.resolver.{ts,js}"],
  emitSchemaFile: true,
  nullableByDefault: true,
  validate: false,
});
