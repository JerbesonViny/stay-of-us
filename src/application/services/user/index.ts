export * from "./find-users.service";
