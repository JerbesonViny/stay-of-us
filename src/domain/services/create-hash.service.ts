export namespace CreateHash {
  export type Input = {
    text: string;
  };
}

export interface CreateHash {
  perform(input: CreateHash.Input): string;
}
