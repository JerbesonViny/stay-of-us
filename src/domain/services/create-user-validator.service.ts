export namespace CreateUserValidatorService {
  export type Input = {
    name: string;
    login: string;
    password: string;
    confirmPassword: string;
  };

  export type ValidatePasswordInput = {
    password: string;
    confirmPassword: string;
  };

  export type CheckUserExistsInput = {
    login: string;
  };
}

export interface CreateUserValidatorService {
  validate(input: CreateUserValidatorService.Input): Promise<boolean>;
}

export const CREATE_USER_VALIDATOR_SERVICE = "create-user-validator.service";
