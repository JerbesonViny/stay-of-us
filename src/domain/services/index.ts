export * from "./create-hash.service";
export * from "./create-user-validator.service";
