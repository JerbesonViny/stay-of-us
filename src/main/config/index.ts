export * from "./constants";
export * from "./module-alias";
export * from "./mongo-connection";
export * from "./server";
export * from "./apollo-server";
