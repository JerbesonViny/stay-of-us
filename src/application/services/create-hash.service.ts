import { createHash } from "crypto";
import { Service } from "typedi";
import { CreateHash } from "@/domain/services";

export const CREATE_HASH_SERVICE = "create-hash.service";

@Service(CREATE_HASH_SERVICE)
export class CreateHashService implements CreateHash {
  private algorithm: string = "sha512";

  perform({ text }: CreateHash.Input): string {
    const hasher = createHash(this.algorithm);

    hasher.update(text);

    return hasher.digest("hex");
  }
}
