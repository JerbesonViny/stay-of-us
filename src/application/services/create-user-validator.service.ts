import { Inject, Service } from "typedi";
import {
  CREATE_USER_VALIDATOR_SERVICE,
  CreateUserValidatorService,
} from "@/domain/services";
import {
  FIND_USER_REPOSITORY,
  FindUserRepository,
} from "@/domain/contracts/repositories";
import { UserAlreadyExists } from "@/domain/errors";

@Service(CREATE_USER_VALIDATOR_SERVICE)
export class CreateUserValidatorServiceImpl
  implements CreateUserValidatorService
{
  constructor(
    @Inject(FIND_USER_REPOSITORY)
    private findUserRepository: FindUserRepository
  ) {}

  async validate(input: CreateUserValidatorService.Input): Promise<boolean> {
    this.checkUserExists(input);

    return true;
  }

  private async checkUserExists({
    login,
  }: CreateUserValidatorService.CheckUserExistsInput): Promise<boolean> {
    const user = await this.findUserRepository.perform({ login });

    if (user) {
      throw new UserAlreadyExists();
    }

    return true;
  }
}
