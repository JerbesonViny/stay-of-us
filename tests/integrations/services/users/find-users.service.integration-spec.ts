import "reflect-metadata";
import "../../../../src/main/config/module-alias";
import dotenv from "dotenv";
dotenv.config();

import { FindUsersService } from "@/application/services";
import {
  MongoConnectionSingleton,
  MongoDatabase,
} from "@/main/config/mongo-connection";
import { Db } from "mongodb";
import { populateDatabase, resetDatabase } from "@/tests/utils/mongo";

describe("FindUsersService", () => {
  let findUsersService: FindUsersService;
  let mongoDatabase: MongoDatabase;
  let mongoConnection: Db;

  beforeAll(async () => {
    mongoDatabase = new MongoConnectionSingleton().getInstance();
    mongoConnection = await mongoDatabase.getConnection();
    await resetDatabase({ mongoConnection });
    await populateDatabase({ mongoConnection });
  });

  beforeEach(async () => {
    findUsersService = new FindUsersService(mongoConnection);
  });

  afterEach(async () => {
    jest.clearAllMocks();
  });

  afterAll(async () => {
    await mongoDatabase.close();
  });

  it("Should get all users", async () => {
    const users = await findUsersService.perform();

    expect(users).toMatchSnapshot();
  });
});
