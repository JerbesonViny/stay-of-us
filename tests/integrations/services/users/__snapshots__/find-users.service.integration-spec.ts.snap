// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FindUsersService Should get all users 1`] = `
Array [
  Object {
    "_id": "65ec9ad1e0e9b4dd1e7d1f8a",
    "name": "Sophie",
  },
  Object {
    "_id": "65ec9b61137345c53e0b4b62",
    "name": "Morris",
  },
]
`;
